src/vm/amd64/JitHelpers_Fast.asm

```diff
@@ -730,124 +730,6 @@ g_pObjectClass      equ     ?g_pObjectClass@@3PEAVMethodTable@@EA
 EXTERN  g_pObjectClass:qword
 extern ArrayStoreCheck:proc
 extern ObjIsInstanceOfNoGC:proc
-
-; TODO: put definition for this in asmconstants.h
-CanCast equ     1
-
-;__declspec(naked) void F_CALL_CONV JIT_Stelem_Ref(PtrArray* array, unsigned idx, Object* val)
-LEAF_ENTRY JIT_Stelem_Ref, _TEXT
```

Comment:
I do not think you want to be removing this asm helper as part of this change. Out of all casting helpers, this helper is the hardest one to remove without introducing regressions. I expect you will see 20%+ regressions in number of micro-benchmarks (collections, etc.) with this change.

Also, the plan we have discussed said that removing asm helpers is going to be a follow up change.

Reply:
It calls `ObjIsInstanceOfNoGC` and the rest of `NoGC` stuff indirectly. 
I figured if we are removing NoGC stuff and asm helpers, this will need to go first.

What would be the 20%?   C++ prologue/epilogue? 
Would that be a problem for other asm helpers?

Reply:
I mean - other NoGC helpers are not called from asm helpers and easier to deal with. `JIT_Stelem_Ref` is harder to tease apart. It kind of designed around having unframed casting helper.

It may be possible to keep `ObjIsInstanceOfNoGC` and just reduce it to a cache lookup, 
But then would it be simpler and cheaper to just move the whole deal to C++ helper that does exactly the same thing.

I am not sure about costs of entering C++ method though.

Reply:
You should measure something like this before/after this change:

```
List<string> s = new List<string>();
s.Add("Hello");
for (int i = 0; < 1000000000; i++)
    s[0] = "World";
```

> It may be possible to keep ObjIsInstanceOfNoGC and just reduce it to a cache lookup

Yes, that sounds better to me.